### PR TITLE
[ru] cs-230-deep-learning-tips-and-tricks

### PR DESCRIPTION
**Pull request by AlexandrParkhomenko**
web format: [Tips and tricks](https://alexandrparkhomenko.github.io/stanford/cs-230/deep-learning-tips-and-tricks/index.html)

### CODE REVIEW

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+**36. [Small, Medium, Large]**
+
+&#10230; [Маленький, Средний, Большой]
+
+<br>
+
+
+**37. [Freezes all layers, trains weights on softmax, Freezes most layers, trains weights on last layers and softmax, Trains weights on layers and softmax by initializing weights on pre-trained ones]**
+
+&#10230; [Замораживает все слои, тренирует веса на softmax, Замораживает большинство слоев, тренирует веса на последних слоях и softmax, Тренирует веса на слоях и softmax инициализирует веса на предварительно обученных]
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:259]:
```suggestion
&#10230; [Замораживает все слои, обучает веса на softmax, Замораживает большинство слоев, дообучает веса на последних слоях и softmax, Дообучает веса на большинстве слоев и softmax]
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+**39. Learning rate ― The learning rate, often noted α or sometimes η, indicates at which pace the weights get updated. It can be fixed or adaptively changed. The current most popular method is called Adam, which is a method that adapts the learning rate.**
+
+&#10230; Скорость обучения ― Скорость обучения, часто обозначаемая α или иногда η, указывает, с какой скоростью обновляются веса. Её можно исправить или адаптивно изменить. Самый популярный в настоящее время метод называется Адам, он адаптирует скорость обучения.
+
+<br>
+
+
+**40. Adaptive learning rates ― Letting the learning rate vary when training a model can reduce the training time and improve the numerical optimal solution. While Adam optimizer is the most commonly used technique, others can also be useful. They are summed up in the table below:**
+
+&#10230; Адаптивная скорость обучения ― Изменение скорости обучения при обучении модели может сократить время обучения и улучшить численное оптимальное решение. Хотя оптимизатор Адама является наиболее часто используемым методом, другие также могут быть полезны. Они приведены в таблице ниже:
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:280]:
```suggestion
&#10230; Адаптивная скорость обучения ― Изменение скорости обучения при обучении модели может сократить время обучения и улучшить численное оптимальное решение. Хотя оптимизатор Adam является наиболее часто используемым, другие также могут быть полезны. Они приведены в таблице ниже:
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+**38. Optimizing convergence**
+
+&#10230; Оптимизация сходимости
+
+<br>
+
+
+**39. Learning rate ― The learning rate, often noted α or sometimes η, indicates at which pace the weights get updated. It can be fixed or adaptively changed. The current most popular method is called Adam, which is a method that adapts the learning rate.**
+
+&#10230; Скорость обучения ― Скорость обучения, часто обозначаемая α или иногда η, указывает, с какой скоростью обновляются веса. Её можно исправить или адаптивно изменить. Самый популярный в настоящее время метод называется Адам, он адаптирует скорость обучения.
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:273]:
```suggestion
&#10230; Скорость обучения ― Скорость обучения, часто обозначаемая α или иногда η, указывает, с какой скоростью обновляются веса. Её можно исправить или адаптивно изменить. Самый популярный в настоящее время метод называется Adam, он адаптирует скорость обучения.
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+**42. [Momentum, Dampens oscillations, Improvement to SGD, 2 parameters to tune]**
+
+&#10230; [Momentum, Гасит колебания, Улучшение SGD, 2 параметра для настройки]
+
+<br>
+
+
+**43. [RMSprop, Root Mean Square propagation, Speeds up learning algorithm by controlling oscillations]**
+
+&#10230; [RMSprop, Среднеквадратичное распространение (Root Mean Square propagation), Ускоряет алгоритм обучения за счет управления колебаниями]
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:301]:
```suggestion
&#10230; [RMSprop, Среднеквадратичное распространение ошибки (Root Mean Square propagation), Ускоряет алгоритм обучения за счет управления колебаниями]
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+**46. Regularization**
+
+&#10230; Регуляризация
+
+<br>
+
+
+**47. Dropout ― Dropout is a technique used in neural networks to prevent overfitting the training data by dropping out neurons with probability p>0. It forces the model to avoid relying too much on particular sets of features.**
+
+&#10230; Прореживание ― Dropout - это метод, используемый в нейронных сетях для предотвращения переобучения обучающих данных путем выпадения нейронов с вероятностью p>0. Это заставляет модель не слишком полагаться на определенные наборы функций.
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:329]:
```suggestion
&#10230; Прореживание ― Dropout - это метод, используемый в нейронных сетях для предотвращения переобучения путем зануления активаций нейронов с вероятностью p>0. Это заставляет модель не слишком полагаться на конкретные наборы функций.
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+**48. Remark: most deep learning frameworks parametrize dropout through the 'keep' parameter 1−p.**
+
+&#10230; Примечание: большинство фреймворков глубокого обучения параметризуют исключение с помощью параметра 'keep' 1−p.
+
+<br>
+
+
+**49. Weight regularization ― In order to make sure that the weights are not too large and that the model is not overfitting the training set, regularization techniques are usually performed on the model weights. The main ones are summed up in the table below:**
+
+&#10230; Регуляризация веса ― Чтобы убедиться, что веса не слишком велики и что модель не переобучается на обучающей выборке, обычно выполняются методы регуляризации для весов модели. Основные из них приведены в таблице ниже:
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:343]:
```suggestion
&#10230; Регуляризация весов ― Чтобы убедиться, что веса не слишком велики и что модель не переобучается на обучающей выборке, обычно выполняются методы регуляризации весов модели. Основные из них приведены в таблице ниже:
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+
+**50. [LASSO, Ridge, Elastic Net, Shrinks coefficients to 0, Good for variable selection, Makes coefficients smaller, Tradeoff between variable selection and small coefficients]**
+
+&#10230; [LASSO, Ridge, Elastic Net, Уменьшает коэффициенты до 0, Подходит для выбора переменных, Делает коэффициенты меньше, Компромисс между выбором переменных и небольшими коэффициентами]
+
+<br>
+
+**51. Early stopping ― This regularization technique stops the training process as soon as the validation loss reaches a plateau or starts to increase.**
+
+&#10230; Ранняя остановка ― Этот метод регуляризации останавливает процесс обучения, как только потеря валидации достигает плато или начинает увеличиваться.
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:356]:
```suggestion
&#10230; Ранняя остановка ― Этот метод регуляризации останавливает процесс обучения, как только функция потерь на валидационной выборке достигает плато или начинает увеличиваться.
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+**53. Good practices**
+
+&#10230; Хорошие практики
+
+<br>
+
+
+**54. Overfitting small batch ― When debugging a model, it is often useful to make quick tests to see if there is any major issue with the architecture of the model itself. In particular, in order to make sure that the model can be properly trained, a mini-batch is passed inside the network to see if it can overfit on it. If it cannot, it means that the model is either too complex or not complex enough to even overfit on a small batch, let alone a normal-sized training set.**
+
+&#10230; Переобучение на небольших пакетах ― При отладке модели часто бывает полезно провести быстрые тесты, чтобы увидеть, есть ли какие-либо серьезные проблемы с архитектурой самой модели. В частности, чтобы убедиться, что модель может быть должным образом обучена, внутри сети передается мини-пакет, чтобы увидеть, может ли он переобучиться. Если это невозможно, это означает, что модель либо слишком сложна, либо недостаточно сложна, чтобы даже переобучиться на небольшой партии, не говоря уже об обучающем наборе нормального размера.
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:377]:
```suggestion
&#10230; Переобучение на небольших пакетах ― При отладке модели часто бывает полезно провести быстрые тесты, чтобы увидеть, есть ли какие-либо серьезные проблемы с архитектурой самой модели. В частности, чтобы убедиться, что модель должным образом обучена, на вход сети передается мини-пакет, чтобы увидеть, случится ли на нем переобучение. В случае если этого не происходит, то означает, что модель либо слишком сложна, либо недостаточно сложна, чтобы даже переобучиться на небольшой партии, не говоря уже об обучающем наборе нормального размера.
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 9]
```diff
@@ -0,0 +1,450 @@
...
+
+<br>
+
+
+**55. Gradient checking ― Gradient checking is a method used during the implementation of the backward pass of a neural network. It compares the value of the analytical gradient to the numerical gradient at given points and plays the role of a sanity-check for correctness.**
+
+&#10230; Проверка градиента ― это метод, используемый во время реализации обратного прохода нейронной сети. Он сравнивает значение аналитического градиента с числовым градиентом в заданных точках и играет роль проверки правильности.
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:384]:
```suggestion
&#10230; Проверка градиента ― это метод, используемый во время реализации обратного прохода нейронной сети. Он сравнивает значение аналитического градиента с числовым градиентом в заданных точках и играет роль проверки реализации на корректность.
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 10]
```diff
@@ -0,0 +1,450 @@
...
+**57. [Formula, Comments]**
+
+&#10230; [Формула, Комментарии]
+
+<br>
+
+
+**58. [Expensive; loss has to be computed two times per dimension, Used to verify correctness of analytical implementation, Trade-off in choosing h not too small (numerical instability) nor too large (poor gradient approximation)]**
+
+&#10230; [Дорого; потери должны вычисляться два раза для каждого измерения, Используется для проверки правильности аналитической реализации, Компромисс при выборе h не слишком малого (числовая нестабильность) или слишком большого (плохое приближение градиента)]
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:405]:
```suggestion
&#10230; [Дорого; потери должны вычисляться два раза для каждого измерения, Используется для проверки правильности аналитической реализации, Компромисс при выборе h не слишком малого (числовая нестабильность), но и не слишком большого (плохое приближение градиента)]
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 11]
```diff
@@ -0,0 +1,450 @@
...
+
+**61. Original authors**
+
+&#10230; Авторы оригинала: Afshine Amidi и Shervine Amidi ― https://github.com/shervinea
+
+<br>
+
+**62. Translated by X, Y and Z**
+
+&#10230; Российская адаптация: Пархоменко Александр ― https://github.com/AlexandrParkhomenko
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:430]:
```suggestion
&#10230; Переведено на русский язык: Пархоменко Александр ― https://github.com/AlexandrParkhomenko
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 12]
```diff
@@ -0,0 +1,450 @@
...
+
+<br>
+
+**63. Reviewed by X, Y and Z**
+
+&#10230; Проверено X, Y и Z
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:436]:
```suggestion
&#10230; Проверено на русском языке: Труш Георгий (Georgy Trush) ― https://github.com/geotrush
```

**Review by tensorush — CHANGES_REQUESTED** [review]
Immaculate translation, which remains faithful to the original! I'm proposing only some minor changes.

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 13]
```diff
@@ -0,0 +1,450 @@
...
+
+
+**60. The Deep Learning cheatsheets are now available in [target language].**
+
+&#10230; Шпаргалки по глубокому обучению теперь доступны в формате [target language].
+
+
+**61. Original authors**
+
+&#10230; Авторы оригинала: Afshine Amidi и Shervine Amidi ― https://github.com/shervinea
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:424]:
```suggestion
&#10230; Авторы оригинала: Afshine Amidi и Shervine Amidi ― https://github.com/afshinea и https://github.com/shervinea
```

File: ru/cs-230-deep-learning-tips-and-tricks.md [hunk 14]
```diff
@@ -0,0 +1,450 @@
...
+**59. ['Exact' result, Direct computation, Used in the final implementation]**
+
+&#10230; ['Точный' результат, Прямое вычисление, Используемое в окончательной реализации]
+
+<br>
+
+
+**60. The Deep Learning cheatsheets are now available in [target language].**
+
+&#10230; Шпаргалки по глубокому обучению теперь доступны в формате [target language].
```
Comment (geotrush) [inline ru/cs-230-deep-learning-tips-and-tricks.md:419]:
```suggestion
&#10230; Шпаргалки по глубокому обучению теперь доступны на русском языке.
```

**Pull request closed without merging**